src/mmfatl.c

```diff
@@ -262,6 +261,19 @@ static void fatalErrorInit(void) {
   initBuffer();
 }
 
+/*!
+ * copy text verbatim from a format string to the message buffer, until either
+ * the format ends, or a placeholder is encountered.
+ * \post member format of \ref state is advanced over the copied text, if no
+ *   overflow.
+ * \post member format of \ref state points to the terminating \ref NUL on
+ *   overflow.
+ */
+static void handleText(void) {
```

Comment:
Why doesn't this have any arguments? This is the kind of C function I hate to see, relying on lots of global state.

Reply:
One could add a pointer to the ParserState, but that is a global instance anyway.  Maybe one can create the instance on the stack, and pass it around, which I would usually do, but we are in a critical and memory tight situation, possibly including the stack, forcing us to use globals.  I'd love to throw an exception to get a clean stack again, but C 99 to my knowledge has no such mechanism.

I'll think it over, whether some minor unforced stack usage is acceptable.

Reply:
I agree with you that generally the signature of a function should expose
- which objects are read
- which are modified.
This is for better readability and maintenance.
In our case objects are limited to two, namely a Buffer and a ParserState.  For the sake of a good coding style I 'localized' the State, but did not do so with Buffer, since this is clearly a singleton.
The State is not likely to appear in many instances, too, or be created on the stack, since it is heavyweight as a container for the values replacing placeholders, but this may change.  The values are a constant block of data, not subject to change, only the current position is important.  So fine-tuning may remove the values from the State in future.